src/app/shared-services/db/sqlite.service.ts

```diff
@@ -85,8 +85,9 @@ export class SqliteService {
     const set = [
       {
         // insert or update trajectory
-        statement:
-          'INSERT OR UPDATE INTO trajectories (id,type,placename,durationDays) VALUES (?,?,?,?)',
+        statement: `INSERT INTO trajectories (id,type,placename,durationDays) VALUES (?,?,?,?)
+          ON CONFLICT(id) DO UPDATE SET type=?, placename=?, durationDays=?;`,
+
         values: [id, type, placename, durationDays].map(normalize),
```

Comment:
> Don't we need to pass 3 more values as well?

You're absolutely right!
Interestingly it works without those values, too. Possibly because there is no _real_ update happening, as those fields are not changed at this point (yet).